features/admin.sms-templates.v1/pages/sms-customization.tsx

```diff
@@ -0,0 +1,457 @@
+
+type SmsCustomizationPageInterface = IdentifiableComponentInterface;
+
+/**
+ * SMS customization page.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns Main Page for SMS Customization.
+ */
+const SmsCustomizationPage: FunctionComponent<SmsCustomizationPageInterface> = (
+    props: SmsCustomizationPageInterface
+): ReactElement => {
+
+    const { ["data-componentid"]: componentId } = props;
+
+    const [ availableSmsTemplatesList, setAvailableSmsTemplatesList ] = useState<SmsTemplateType[]>([]);
+    const [ currentSmsTemplate, setCurrentSmsTemplate ] = useState<SmsTemplate>();
+    const [ isSystemTemplate, setIsSystemTemplate ] = useState(true);
+    const [ isTemplateNotAvailable, setIsTemplateNotAvailable ] = useState(false);
+    const [ selectedLocale, setSelectedLocale ] = useState(I18nConstants.DEFAULT_FALLBACK_LANGUAGE);
+    const [ selectedSmsTemplateId, setSelectedSmsTemplateId ] = useState<string>();
+    const [ selectedSmsTemplateDescription, setSelectedSmsTemplateDescription ] = useState<string>();
+    const [ selectedSmsTemplate, setSelectedSmsTemplate ] = useState<SmsTemplate>();
+
+    const smsTemplates: Record<string, string>[] = useSelector(
+        (state: AppState) => state.config.deployment.extensions.smsTemplates) as Record<string, string>[];
+    const enableCustomSmsTemplates: boolean = useSelector(
+        (state: AppState) => state?.config?.ui?.enableCustomSmsTemplates);
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const smsFeatureConfig: FeatureAccessConfigInterface = useSelector(
+        (state: AppState) => state.config.ui.features.smsTemplates);
+
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+    const hasUpdatePermission : boolean = useRequiredScopes(smsFeatureConfig?.scopes?.update);
+    const hasCreatePermission : boolean = useRequiredScopes(smsFeatureConfig?.scopes?.create);
+
+    const isReadOnly: boolean = useMemo(() => {
+        return !smsFeatureConfig.enabled || !hasUpdatePermission;
+    }, [ smsFeatureConfig, allowedScopes ]);
+
+    const hasSmsTemplateCreatePermissions: boolean = useMemo(() => {
+        return smsFeatureConfig.enabled && hasCreatePermission;
+    }, [ smsFeatureConfig, allowedScopes ]);
+
+    const {
+        data: smsTemplatesList,
+        isLoading: isSmsTemplatesListLoading,
+        error: smsTemplatesListError
+    } = useSmsTemplatesList();
+
+    const {
+        data: smsTemplate,
+        isLoading: isSmsTemplateLoading,
+        error: smsTemplateError,
+        mutate: smsTemplateMutate
+    } = useSmsTemplate(selectedSmsTemplateId, selectedLocale, setIsSystemTemplate);
+
+    useEffect(() => {
+        // we don't have a good displayName and description coming from the backend
```

Comment:
Is this already tracked as a BE improvement?

Reply:
@RushanNanayakkara

Reply:
Our template data models do not included a description. This will need db level changes and a migration. Therefore, needs to be discussed before implementation. Let's create an issue to track this seperately.